File: solutions/R/inference_and_estimation_ex4.R
# Using the power.t.test function
# 80% power
result_80 <- power.t.test(
  delta = 0.4, # expected difference
  sd = 1.2, # standard deviation
  sig.level = 0.05,
  power = 0.80,
  type = "two.sample",
  alternative = "two.sided"
)
cat("Sample size per group (80% power):", ceiling(result_80$n), "\n")

# 90% power
result_90 <- power.t.test(
  delta = 0.4,
  sd = 1.2,
  sig.level = 0.05,
  power = 0.90,
  type = "two.sample",
  alternative = "two.sided"
)
cat("Sample size per group (90% power):", ceiling(result_90$n), "\n")

cat(
  "\nIncreasing power from 80% to 90% requires about",
  round((ceiling(result_90$n) / ceiling(result_80$n) - 1) * 100),
  "% more participants per group.\n"
)

File: solutions/python/inference_and_estimation_ex4.py
from statsmodels.stats.power import TTestIndPower

analysis = TTestIndPower()

# Cohen's d = delta / sd = 0.4 / 1.2
effect_size = 0.4 / 1.2

# 80% power
n_80 = analysis.solve_power(effect_size=effect_size, alpha=0.05, power=0.80,
                            alternative='two-sided')
print(f"Sample size per group (80% power): {int(np.ceil(n_80))}")

# 90% power
n_90 = analysis.solve_power(effect_size=effect_size, alpha=0.05, power=0.90,
                            alternative='two-sided')
print(f"Sample size per group (90% power): {int(np.ceil(n_90))}")

print(f"\nIncreasing from 80% to 90% power requires about "
      f"{int(round((n_90/n_80 - 1) * 100))}% more participants per group.")